### DISCUSSION TITLE
在Action列表写Javascript Transformer，无法使用函数。

### DISCUSSION THREAD

**Discussion by MasterZZJ** [illacloud/illa-builder#545] [Q&A]
### 我想要对在数据库查询到的内容进行修改再返回，例如字符型变为数值型，数值型变为布尔型。

例如：
我想要将字符型的"0"或"1"，通过`Number()`函数转换成数值。但结果并不尽如人意。会发生错误，并且数据库下面的Transformer会整体无效，直接返回初始的data。

### 当我判断字符串相等时总会出现问题

例如下段代码：

```JS
// data[0]中的active属性值是字符型的变量，取值为"0"或"1"
if (data[0].active=="0")
    return false;
return true;
```

由于函数会出错，不能使用。所以我想出了这种方法，通过判断字符串相等来决定返回值，但实践证明这样也不可以。请问是有什么bug嘛？或者是我才疏学浅哪里操作不当？还是我逻辑有错误？另外，如果想实现数据类型的转换，正确的操作方式是什么，请大佬帮帮忙！万分感谢！！！

**Comment by AruSeito** [comment]
感谢您的反馈，确实存在 `return false`的时候会引发无法出现运行结果的问题。

如果您要修改内容再返回，可以尝试如下代码。

```js
return data.map(v=>{
  return v.active == "0" ? {...v,active:false}:{...v,active:true}
})
```
